src/Build/Construction/ProjectElementContainer.cs

```diff
@@ -749,6 +753,52 @@ public IEnumerator<ProjectElement> GetEnumerator()
                 return _enumerator;
             }
 
+            void ICollection<ProjectElement>.Add(ProjectElement item)
+            {
+                throw new NotSupportedException();
+            }
+
+            void ICollection<ProjectElement>.Clear()
+            {
+                throw new NotSupportedException();
+            }
+
+            bool ICollection<ProjectElement>.Contains(ProjectElement item)
+            {
+                _enumeratorAux.Reset();
+                while (_enumeratorAux.MoveNext())
+                {
+                    if (item.Equals(_enumeratorAux.Current))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            void ICollection<ProjectElement>.CopyTo(ProjectElement[] array, int arrayIndex)
+            {
+                _enumeratorAux.Reset();
+                while (arrayIndex < array.Length && _enumeratorAux.MoveNext())
+                {
+                    array[arrayIndex++] = _enumeratorAux.Current;
+                }
+            }
+
+            private int GetCount()
+            {
+                if (_count == -1)
+                {
+                    _count++;
+                    _enumeratorAux.Reset();
```

Comment:
Should `_enumeratorAux` be reset after each use? It looks like you're resetting it before each use so this may be redundant.

Reply:
Putting it at the beginning or the end should be equivalent, but putting it at the beginning ensures you're in the right state when you start, which would help pin down the error should you forget it in adding new code.